src/openms/include/OpenMS/ANALYSIS/ID/SiriusAdapterAlgorithm.h

```diff
@@ -43,6 +43,8 @@
 #include <OpenMS/SYSTEM/File.h>
 #include <OpenMS/FORMAT/FeatureXMLFile.h>
 
+#include <QDir>
```

Comment:
you can move most includes to the .cpp and just forward declare arguments to the major functions

Reply:
Thanks for all your suggestions! 

Could you have a third look on the forward declarations - if I did that correctly?